src/service/document-info-impl.js

```diff
@@ -73,13 +78,31 @@ export class DocInfo {
           : sourceUrl;
     }
     const pageViewId = getPageViewId(ampdoc.win);
+
+    const doc = ampdoc.win.document;
+    if (doc.head) {
+      for (let n = doc.head.firstElementChild; n; n = n.nextElementSibling) {
+        if (n.tagName != 'LINK') {
+          continue;
+        }
+        const rel = n.getAttribute('rel');
+        const href = n.getAttribute('href');
+        if (!rel || !href || rel == 'prefetch' || rel == 'preload' ||
+            rel == 'preconnect') {
+          continue;
+        }
+        linkRels[rel] = parseUrl(href).href;
```

Comment:
Or `!Object<string, <string> | !Array<string>>` ? If there's only one value, make it a single string?